java/jshell.support/src/org/netbeans/modules/jshell/editor/CompletionFilter.java

```diff
@@ -55,6 +61,29 @@
  */
 final class CompletionFilter extends DocTrees {
 
+    private static final Logger LOG = Logger.getLogger(CompletionFilter.class.getName());
+
+    private static final MethodHandle MH_GET_TYPE;
+    private static final MethodHandle MH_GET_CHARACTERS;
+
+    static {
+        MethodHandle getTypeBuilder = null;
+        MethodHandle getCharactersBuilder = null;
+        Lookup lookup = MethodHandles.lookup();
+        try {
+            getTypeBuilder = lookup.findVirtual(DocTrees.class, "getType", MethodType.methodType(TypeMirror.class, DocTreePath.class));
```

Comment:
I'd suggest a more prominent LOG message, as `getType()` method will NPE if this `findVirtual` fails.
I am not very familiar with shimming different javac versions - @jlahoda should definitely review this.

Reply:
Raised log level to warning - in a normal situation `DocTrees` is loaded from nb-javac and thus the two methods are present and can be resolved here. If it fails the methods are not present and thus can't be called in the first place, so I think this should be safe.